es714x/src/main/scala/tech/beshu/ror/es/actions/rrtestsettings/RRTestSettingsResponse.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.es.actions.rrtestsettings
+
+import org.apache.logging.log4j.scala.Logging
+import org.elasticsearch.action.ActionResponse
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.xcontent.{ToXContent, ToXContentObject, XContentBuilder}
+import tech.beshu.ror.api.TestSettingsApi
+import tech.beshu.ror.api.TestSettingsApi.TestSettingsResponse._
+
+import scala.collection.JavaConverters._
+
+class RRTestSettingsResponse(response: TestSettingsApi.TestSettingsResponse)
+  extends ActionResponse with ToXContentObject with Logging {
+
+  override def toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder = {
+    response match {
+      case res: ProvideTestSettings.CurrentTestSettings => currentSettingsJson(builder, res)
+      case ProvideTestSettings.TestSettingsNotConfigured(message) => addResponseJson(builder, response.status, message)
+      case res: ProvideTestSettings.TestSettingsInvalidated => invalidatedSettingsJson(builder, res)
+      case UpdateTestSettings.SuccessResponse(message) => addResponseJson(builder, response.status, message)
+      case UpdateTestSettings.FailedResponse(message) => addResponseJson(builder, response.status, message)
+      case InvalidateTestSettings.SuccessResponse(message) => addResponseJson(builder, response.status, message)
+      case res: ProvideLocalUsers.SuccessResponse => provideLocalUsersJson(builder, res)
+      case ProvideLocalUsers.TestSettingsNotConfigured(message) => addResponseJson(builder, response.status, message)
+      case Failure(message) => addResponseJson(builder, response.status, message)
```

Comment:
`ConfigApi.Failure` approach is a little legacy, I think. Here we should return 400 or 500 depending on the context:
1. malformed request
2. internal problem

Reply:
Removed

Reply:
but I'm not sure if we do return 400 for the malformed body (the exception case looks good)

Reply:
Status 400 is now returned